pkg/filesystem/subvolume.go

```diff
@@ -52,6 +58,71 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	resource := "cephclusters"
+	cephclusters, err := clientsets.ListResourcesDynamically(ctx, "ceph.rook.io", "v1", resource, clusterNamespace)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	if len(cephclusters) == 0 {
+		logging.Info("resource %s was not found on the cluster", resource)
+	}
+
+	for _, cephcluster := range cephclusters {
+		spec, _, err := unstructured.NestedMap(cephcluster.Object, "spec", "external")
+		if err != nil {
+			fmt.Printf("Error accessing 'externalStorage': %v\n", err)
+			os.Exit(1)
+		}
+		if spec["enable"] != nil {
+			enable = spec["enable"].(bool)
+			return enable
+		}
+	}
+	return enable
+}
+
+// getExternalClusterDetails exports the CEPH_ARGS with required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	var admin_id, admin_key, m string
+
+	scList, err := clientsets.Kube.CoreV1().Secrets(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+	}
+	for _, sc := range scList.Items {
+		if sc.ObjectMeta.Name == "rook-csi-cephfs-node" {
+			data := sc.Data
+			admin_id = string(data["adminID"])
+			admin_key = string(data["adminKey"])
+		}
+	}
+
+	cmList, err := clientsets.Kube.CoreV1().ConfigMaps(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching configmaps: %v\n", err))
+	}
+	for _, cm := range cmList.Items {
+		if cm.ObjectMeta.Name == "rook-ceph-mon-endpoints" {
+			// data is of the below format
+			// "ceph-ci-vavuthuextdevyp2-btr2b-hwxgpn-node1-installer=10.0.209.241:6789,
+			// ceph-ci-vavuthuextdevyp2-btr2b-hwxgpn-node2=10.0.210.220:6789,
+			// ceph-ci-vavuthuextdevyp2-btr2b-hwxgpn-node5=10.0.209.2:6789,
+			// ceph-ci-vavuthuextdevyp2-btr2b-hwxgpn-node3=10.0.211.109:6789"
```

Comment:
can we use `csi-cluster-config-json` from the `"rook-ceph-mon-endpoints"` or use `rook-ceph-csi-config` configmap as its in known format and we don't need to split anything, just need to unmarshal to a struct